hack/vmimage.sh

```diff
@@ -35,6 +36,15 @@ make create
 
 make e2e
 
+# Run the image validation and capture the artifacts
+go generate ./... && go run -ldflags "-X main.gitCommit=$GITCOMMIT" ./cmd/vmimage -buildResourceGroup "${BUILD_RESOURCE_GROUP}" -imageResourceGroup "${IMAGE_RESOURCEGROUP}" -image "${IMAGE_RESOURCENAME}" -imageStorageAccount "${IMAGE_STORAGEACCOUNT}" -validate -preserveBuildResourceGroup
+
+# Currently there are only 3 logs we want to capture
+mv /tmp/{yum_update_info,yum_check_update,scap-report.html} ${ARTIFACTS:-/tmp}/ || true
+
+# delete the image validation build_resource_group
+az group delete -g "${BUILD_RESOURCE_GROUP}" --yes --no-wait
```

Comment:
can you move this line into the cleanup function at the top to make sure `$BUILD_RESOURCE_GROUP` is cleaned up if an error occurs during `./cmd/vmimage`?

Reply:
I think that should work.